tests/certification/secretstores/azure/keyvault/keyvault_test.go

```diff
@@ -98,4 +103,92 @@ func TestKeyVault(t *testing.T) {
 			))).
 		Step("Getting known secret", testGetKnownSecret, sidecar.Stop(sidecarName)).
 		Run()
+
+	// This test reuses the Azure conformance test resources created using
+	// .github/infrastructure/conformance/azure/setup-azure-conf-test.sh
+	managedIdentityTest := func(ctx flow.Context) error {
+		_, err := exec.LookPath("az")
+		if err != nil {
+			t.Error("azure-cli not installed")
+		}
+		groupName, envVarFound := os.LookupEnv("AzureResourceGroupName")
+		if envVarFound == false {
+			t.Error("AzureResourceGroupName environment variable not set")
+		}
+
+		managedIdentityQuery := fmt.Sprintf("az identity show -g %s -n azure-managed-identity -otsv --query id", groupName)
+		managedIdentityID, err := exec.Command("bash", "-c", managedIdentityQuery).Output()
+		if err != nil {
+			t.Error("azure-managed-identity not found")
+		}
+
+		registryName, envVarFound := os.LookupEnv("AzureContainerRegistryName")
+		if envVarFound == false {
+			t.Error("AzureContainerRegistryName environment variable not set")
+		}
+		keyvaultName, envVarFound := os.LookupEnv("AzureKeyVaultName")
+		if envVarFound == false {
+			t.Error("AzureKeyVaultName environment variable not set")
+		}
+
+		// build and publish the image
+		buildContainerCommand := fmt.Sprintf("az acr build --image certification/keyvault:latest --registry %s --file managed-identity-app/Dockerfile ./managed-identity-app", registryName)
+		fmt.Printf("Building Container ==== : %s\n", buildContainerCommand)
+		err = exec.Command("bash", "-c", buildContainerCommand).Run()
+		if err != nil {
+			t.Error("failed to build and publish container")
+		}
+
+		// generate random string
+		rand.Seed(time.Now().UnixNano())
+		containerName := fmt.Sprintf("managedidentitytest%d", rand.Intn(10000)+1)
+
+		// run the container using Azure Container Instances with injected managed identity
+		registryPasswordSubcommand := fmt.Sprintf("az acr credential show -n %s --query passwords[0].value -otsv", registryName)
+		registryUsernameSubcommand := fmt.Sprintf("az acr credential show -n %s --query username -otsv", registryName)
+		containerCreateCommand := fmt.Sprintf("az container create -g %s -n %s --image %s.azurecr.io/certification/keyvault:latest --restart-policy never --environment-variables AzureKeyVaultName=%s --registry-password $(%s) --registry-username $(%s) --assign-identity %s", groupName, containerName, registryName, keyvaultName, registryPasswordSubcommand, registryUsernameSubcommand, managedIdentityID)
```

Comment:
For my understanding, why is ACR needed at all? `az container create` supports taking a local image in my experience, and it looks like the injection of the managed identity is unrelated to the origin of the image?